iam.tf

```diff
@@ -27,13 +27,13 @@ data "aws_iam_policy_document" "route53" {
 
 resource "aws_iam_policy" "route53" {
   count  = var.route53_hosted_zone_name != "" ? 1 : 0
-  name   = "${var.deploy_id}-Route53"
+  name   = "${var.deploy_id}-route53"
   path   = "/"
   policy = data.aws_iam_policy_document.route53[0].json
 }
 
 resource "aws_iam_role_policy_attachment" "route53" {
-  for_each   = var.route53_hosted_zone_name != "" ? toset([for r in module.eks.eks_node_roles : r.name]) : []
+  count      = var.route53_hosted_zone_name != "" ? length(module.eks.eks_node_roles) : 0
   policy_arn = aws_iam_policy.route53[0].arn
-  role       = each.value
+  role       = lookup(module.eks.eks_node_roles[count.index], "name")
```

Comment:
This was to fix the fact that terraform refresh can fail here under some circumstances.

Reply:
Just curious, what does this change that allows it to succeed? Or, what fails during refresh?

Reply:
This is the error:

```
The "for_each" value depends on resource attributes that cannot be determined
until apply, so Terraform cannot predict how many instances will be created.
To work around this, use the -target argument to first apply only the
resources that the for_each depends on.
```

@miguelhar said that count fixes it and gave me the snippet, and I was able to confirm it did. These failures happen when things are partially provisioned and you're retrying, which I just happened to run into while figuring out what permissions were missing (thing provisions -> fix iam -> can't continue because refresh fails).